src/codewhisperer/util/runtimeLanguageContext.ts

```diff
@@ -193,12 +211,21 @@ export class RuntimeLanguageContext {
     /**
      *
      * @param languageId: either vscodeLanguageId or CodewhispererLanguage
-     * @returns ture if the language is supported by CodeWhisperer otherwise false
+     * @returns true if the language is supported by CodeWhisperer otherwise false
      */
     public isLanguageSupported(languageId: string): boolean {
         const lang = this.normalizeLanguage(languageId)
         return lang !== undefined && this.normalizeLanguage(languageId) !== 'plaintext'
     }
+    /**
+     *
+     * @param fileFormat : vscode editor filecontext filename extension
+     * @returns  true if the fileformat is supported by CodeWhisperer otherwise false
+     */
+    public isFileFormatSupported(fileFormat: string): boolean {
+        const fileformat = this.supportedLanguageExtensionMap.get(fileFormat)
+        return fileformat !== undefined && this.supportedLanguageExtensionMap.get(fileformat) !== 'txt'
```

Comment:
`this.supportedLanguageExtensionMap.get(fileFormat)` is doing a lookup on the thing it just looked up. is that intentional?

Reply:
prefer having single one interface `isLanguageSupported` but we can make the signature to be `isLanguageSupported(documetn: vscode.TextDocument)`

Reply:
> this.supportedLanguageExtensionMap.get(fileFormat) is doing a lookup on the thing it just looked up. is that intentional?

I think we did not check for file extension before 

```
    (runtimeLanguageContext.isLanguageSupported(req.fileContext.programmingLanguage.languageName) ||
             runtimeLanguageContext.isFileFormatSupported(
                 req.fileContext.filename.substring(req.fileContext.filename.lastIndexOf('.') + 1)
             ))
```

In this code we check for language and then we check for file extension if anything matches this is treated as a validLanguage name

Reply:
correct, we didn't check file ext before, I mean can we add these layer of logic into existing `isLanguageSupport()` method instead exposing a new API? 

```
public isLanguageSupport(document: vscode.TextDocument) {
    const Lang = this.normalizedLanguage(document.languageId)
    if (this.supportedLanguageMap.get(normalizedLang)) {
        return true
    } else {
         // check file ext
    }
}
```

Reply:
the reasoning is that basically we will always need to 
1. check language id if present
2. if step 1 fails,  then go check file ext.

in this case we don't really split these 2 into 2 API and callers have to explicitly call these 2 APIs. 1 is enough and should be more error prone IMO.

Reply:
Yes we can add this to `isLanguageSupport` but as I mentioned below we need more refactoring of the code in different files. Let me take up and do this in this PR itself